securedrop-admin

```diff
@@ -50,42 +50,119 @@ def sdconfig(args):
                             cwd=ANSIBLE_PATH, env=os.environ.copy())
 
 
+def run_command(command):
+    """
+    Wrapper function to display stdout for running command,
+    similar to how shelling out in a Bash script displays rolling output.
+
+    Yields a list of the stdout from the `command`, and raises a
+    CalledProcessError if `command` returns non-zero.
+    """
+    popen = subprocess.Popen(command, stdout=subprocess.PIPE)
+    for stdout_line in iter(popen.stdout.readline, ""):
+        yield stdout_line
+    popen.stdout.close()
+    return_code = popen.wait()
+    if return_code:
+        raise subprocess.CalledProcessError(return_code, command)
+
+
+def install_apt_dependencies(args):
+    """
+    Install apt dependencies in Tails. In order to install Ansible in
+    a virtualenv, first there are a number of Python prerequisites.
+    """
+    sdlog.info("Installing SecureDrop Admin dependencies")
+    sdlog.info(("You'll be prompted for the temporary Tails admin password,"
+                " which was set on Tails login screen"))
+
+    apt_command = ['sudo', 'su', '-c',
+                   "apt-get update && \
+                   apt-get install -y \
+                   python-virtualenv \
+                   python-pip \
+                   ccontrol \
+                   virtualenv \
+                   libffi-dev \
+                   libssl-dev \
+                   libpython2.7-dev",
+                   ]
+
+    try:
+        # Print command results in real-time, to keep Admin apprised
+        # of progress during long-running command.
+        for output_line in run_command(apt_command):
+            print(output_line.rstrip())
+    except subprocess.CalledProcessError:
+        # Tails supports apt persistence, which was used by SecureDrop
+        # under Tails 2.x. If updates are being applied, don't try to pile
+        # on with more apt requests.
+        sdlog.error(("Cannot lock apt. Tails may be performing automatic"
```

Comment:
Sooo I don't think this specific messaging makes sense here -- for example, I purposely killed the `apt-get update` process while it was running, and that output looks like this:

```
INFO: You'll be prompted for the temporary Tails admin password, which was set on Tails login screen
[sudo] password for amnesia: 
Hit:1 tor+http://sdscoq7snqtznauu.onion/torproject.org stretch InRelease
Hit:2 tor+http://sdscoq7snqtznauu.onion/torproject.org sid InRelease
Ign:3 tor+http://vwakviie2ienjx6t.onion/debian stretch InRelease
Hit:4 tor+http://vwakviie2ienjx6t.onion/debian sid InRelease
Hit:5 tor+http://vwakviie2ienjx6t.onion/debian stretch Release
Hit:7 tor+http://jenw7xbd6tf7vfhp.onion 3.0 InRelease
Reading package lists...
Reading package lists...
ERROR: Cannot lock apt. Tails may be performing automatic updates. Please try again in a few minutes.
```
where as the behavior when apt is locked looks like this:

```
INFO: You'll be prompted for the temporary Tails admin password, which was set on Tails login screen
[sudo] password for amnesia: 
Reading package lists...
E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/
ERROR: Cannot lock apt. Tails may be performing automatic updates. Please try again in a few minutes.
```
I really don't anticipate other ways in which apt would bomb out besides these two scenarios -- I tried disconnecting the network but it took forrrrrever to bomb out and didnt have patience to see the full output. So I dunno, I'm just bringing this up for discussion -- should the error output be more of a generic `apt failed` or are we just going to assume that apt will always bomb out because it is locked? Really I dont think this is enough to block merge of this great PR. So I'm still going to approve as-is.